src/lexer.coffee

```diff
@@ -1005,12 +1023,26 @@ exports.Lexer = class Lexer
   # Helpers
   # -------
 
+  # Compensate for the things we strip out initially (e.g. carriage returns)
+  # so that location data stays accurate with respect to the original source file.
+  getLocationDataCompensation: (start, end) ->
```

Comment:
Then here is a helper to calculate how much total compensation (ie adjustment) should be made for a given offset range